### PR TITLE
PR - Backend - Driftkart (E-commerce App)

### PR DESCRIPTION
**Pull request by duhbhavesh**
Hello, Please review my PR for the E-commerce Backend.

_Thank you._

### CODE REVIEW

File: server/db/db.connect.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const mongoose = require('mongoose');
+const toJson = require('@meanie/mongoose-to-json');
+mongoose.plugin(toJson);
+
```
Comment (shubham-singh) [inline server/db/db.connect.js:4]: what is this toJson plugin for?

File: server/routes/cart-router.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const express = require('express');
+const {
+   getCartItems,
+   addCartItem,
+   deleteCartItem,
+} = require('../controllers/cart');
+const router = express.Router();
+
+router.route('/').get(getCartItems).post(addCartItem);
+
```
Comment (shubham-singh) [inline server/routes/cart-router.js:10]: You should write .get(getCartItems) in a new line and .post(addCartItem) in the next one. It's just nitpicking but it easier to read and intuitive

File: server/models/cart.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const mongoose = require('mongoose');
+const { Schema } = mongoose;
+
+const CartItemSchema = new Schema({
+   _id: { type: Schema.Types.ObjectId, ref: 'Product' },
+   quantity: Number,
+});
+
+const CartItem = mongoose.model('CartItem', CartItemSchema);
+
```
Comment (shubham-singh) [inline server/models/cart.js:10]: you want to use cartitems as your table name for cart?
carts sounds good too. maybe look into that.

same with wishlist
(this is your personal preference)

File: server/models/products.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      productsData.forEach(async (product) => {
+         const newProduct = new Product(product);
+         const savedProduct = await newProduct.save();
+         console.log(savedProduct);
+      });
+   } catch (error) {
+      console.log(error);
+   }
+};
+
```
Comment (shubham-singh) [inline server/models/products.js:30]: extract this PopulateProducts function in controller/products as well.

**Review by shubham-singh — COMMENTED** [review]
Hi, @duhbhavesh ,
The is well written code. I have some minor suggestions, maybe look into that. Rest all good!

File: server/db/db.connect.js (hunk 1 above)
Comment (duhbhavesh) [reply server/db/db.connect.js:4]: toJson plugin is used to get the normalized JSON output.
Mongodb by default it outputs `id` as `_id` and if we want normalized output like `id` you can use this plugin. Now you don't have to change the code to `id` to `_id` as your default JSON output will be `id` only. 
Hope it makes sense.

Here is the reference - [link](https://www.npmjs.com/package/@meanie/mongoose-to-json)

File: server/routes/cart-router.js (hunk 2 above)
Comment (duhbhavesh) [reply server/routes/cart-router.js:10]: LOL that's prettier's doing. I did the same as you mentioned on repl but prettier wants the code to look like this😂

**Comment by shubham-singh** [comment]
Hi, @duhbhavesh 

This all looks good!!

**Review by rkjain119 — COMMENTED** [review]
Hi @duhbhavesh 
You have written clean code I don't see any problems

File: server/routes/cart-router.js (hunk 2 above)
Comment (jay-rajput30) [reply server/routes/cart-router.js:10]: prettier doesnt makes it look prettier i guess :rofl:

**Comment by jay-rajput30** [comment]
looks good to merge

**Pull request merged**